### PR TITLE
Move Talk creation into own controller

### PR DESCRIPTION
**Pull request by heiglandreas**
This PR moves creation of a talk into it's own controller.

This PR superseeds PR #186 and is Part 1 of at least 7

### CODE REVIEW

File: src/controllers/TalksController.php [hunk 1]
```diff
@@ -98,21 +103,26 @@ public function postAction($request, $db) {
                                 $this->config['akismet']['apiKey'],
                                 $this->config['akismet']['blog']
                             );
-                            $isValid = $spamCheckService->isCommentAcceptable(
+                            $isValid          = $spamCheckService->isCommentAcceptable(
```
Comment (akrabat) [inline src/controllers/TalksController.php]: whitespace change not needed.

File: src/config/routes/2.1.json [hunk 2]
```diff
@@ -51,7 +57,7 @@
     {
         "path": "/talks(/[^/]+)*/?$",
         "controller": "TalksController",
-        "action": "postAction",
+        "action": "createTalkAction",
```
Comment (akrabat) [inline src/config/routes/2.1.json]: This route needs go to postAction as otherwise, I can't create a comment…

Comment (heiglandreas) [inline src/config/routes/2.1.json]: You're right! Sorry, missed that bit...

File: src/controllers/TalksController.php (hunk 1 above)
Comment (heiglandreas) [inline src/controllers/TalksController.php]: Shouldn't be there at all... Will be removed!

File: src/controllers/TalksController.php [hunk 3]
```diff
@@ -123,35 +133,179 @@ public function postAction($request, $db) {
...
+     * @return array|bool
+     */
+    public function createTalkAction($request, $db)
+    {
+        // Set the event-ID either via URL or via POST-Data depending on how the
+        // method was called.
+        // When it's called via a POST to /talks/ the event-id is part of the
+        // POST-data, when it's called via POST to /events/<id>/talks/ the
+        // event-id is part of the URL but will be overwritten by POST-data
+        $event_id = null;
```
Comment (akrabat) [inline src/controllers/TalksController.php]: The documentation says:

> To create a talk, you actually need to make a POST request to the *talks* collection of the *event* that the session belongs to

i.e. POSTing to /talks/ doesn't create a new talk. Do we want it to?

File: src/controllers/TalksController.php [hunk 4]
```diff
@@ -150,10 +150,151 @@ public function postAction($request, $db) {
                     throw new Exception("Operation not supported, sorry", 404);
             }
         } else {
-            throw new Exception("method not yet supported - sorry");
+            $this->createTalkAction($request, $db);
```
Comment (akrabat) [inline src/controllers/TalksController.php]: We don't POST to `/talks` to create a new talk. Please put Exception back.

File: src/inc/Request.php [hunk 5]
```diff
@@ -99,6 +99,9 @@ public function __construct($config, array $server, $parseParams = true)
         }
 
         $this->setBase($this->getScheme() . $this->getHost());
+        if (isset($server['HTTP_PORT']) && $server['HTTP_PORT'] && $server['HTTP_PORT'] != 80) {
+            $this->setBase($this->getBase() . ':' . $server['HTTP_PORT']);
+        }
```
Comment (akrabat) [inline src/inc/Request.php]: Why is this part of this PR?

**Comment by akrabat** [comment]
As this PR moves the talk creation code, can you also remove the old code at https://github.com/joindin/joindin-api/blob/master/src/controllers/EventsController.php#L149-L223 please.

Also, the result of PR shouldn't change functionality.

**Comment by heiglandreas** [comment]
I'll split this one up into two:

* Move stuff from EventsController to TalksController (including removal of old code) without changing functionality
* Altering the functionality as it will be needed to comply with editing talks (Might also go into the PR that handles the editing.

As I'd like to include Lornas Removal of the ```handle```-function I think it'S easier to do a new PR based on the current master instead of modifying this one which will become messy.

**Pull request closed without merging**